src/Common/AzurePSCmdlet.cs

```diff
@@ -434,6 +438,37 @@ protected bool IsVerbose()
             return verbose;
         }
 
+        protected void WriteSurvey()
+        {
+            HostInformationMessage msg = new HostInformationMessage()
+            {
+                Message = string.Format("Survey: How was your experience using {0}.{1}?`\nRun ", ModuleName, ModuleVersion),
+                NoNewLine = true
+            };
+            HostInformationMessage msg2;
+            try
+            {
+                msg2 = new HostInformationMessage()
+                {
+                    Message = "Open-AzSurveyLink",
+                    NoNewLine = true,
+                    ForegroundColor = (ConsoleColor)Host.PrivateData.Properties.Match("ProgressBackgroundColor").SingleOrDefault().Value
```

Comment:
Get color once in one process. If there is exception, fallback to color of warning message?
After it, you don't need to try-catch in each cmdlet.